### PR TITLE
Fargate

### PR DESCRIPTION
**Pull request by ferruzzi**
Added EKS support for Fargate Profiles with unit testing:
- list_fargate_profiles
- create_fargate_profile
- describe_fargate_profile
- delete_fargate_profile

### CODE REVIEW

File: moto/eks/models.py [hunk 1]
```diff
@@ -498,6 +675,51 @@ def validate_launch_template_combination(disk_size, remote_access):
...
+        # Every existing Selector must have a namespace
+        if "namespace" not in selector:
+            raise_exception(message=FARGATE_PROFILE_SELECTOR_NEEDS_NAMESPACE)
+        # If a selector has labels, it can not have more than 5
+        if len(selector.get("labels", {})) > 5:
+            raise_exception(message=FARGATE_PROFILE_TOO_MANY_LABELS)
+
+
+def generate_uuid():
+    return "-".join([random_string(_) for _ in [8, 4, 4, 4, 12]]).lower()
```
Comment (bblommers) [inline moto/eks/models.py:720]: Can we use the build-in method here?
```
from uuid import uuid4
return str(uuid4())
```

File: tests/test_eks/test_eks.py [hunk 2]
```diff
@@ -797,6 +847,490 @@ def test_create_nodegroup_handles_launch_template_combinations(
...
+    FargateProfileBuilder,
+):
+    _, generated_test_data = FargateProfileBuilder()
+
+    result_time = iso_8601_datetime_without_milliseconds(
+        generated_test_data.fargate_describe_output[FargateProfileAttributes.CREATED_AT]
+    )
+
+    if settings.TEST_SERVER_MODE:
+        RegExTemplates.ISO8601_FORMAT.match(result_time).should.be.true
```
Comment (bblommers) [inline tests/test_eks/test_eks.py:1042]: The sure-library has a match-method, that we can use AFAIK:
```
RegExTemplates.ISO8601_FORMAT.should.match(result_time)
```
Slightly easier to read.

File: tests/test_eks/test_eks_constants.py [hunk 3]
```diff
@@ -49,6 +51,9 @@
 NODEROLE_ARN_KEY = "nodeRole"
 NODEROLE_ARN_VALUE = "arn:aws:iam::123456789012:role/role_name"
 
+POD_EXECUTION_ROLE_ARN_KEY = "podExecutionRoleArn"
+POD_EXECUTION_ROLE_ARN_VALUE = "arn:aws:iam::123456789012:role/role_name"
```
Comment (bblommers) [inline tests/test_eks/test_eks_constants.py:55]: The account ID is centralized, so we should use that over hardcoded values where possible:
```
from moto.core import ACCOUNT_ID
```
(Same for `NODEROLE_ARN_VALUE `, actually - I must have missed that in the last review)

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @ferruzzi, thanks for the PR!
You might have to rebase, there's a merge conflict that's stopping the CI from running

File: tests/test_eks/test_eks.py (hunk 2 above)
Comment (ferruzzi) [reply tests/test_eks/test_eks.py:1042]: Neat, I didn't know that one.  Thanks!

**Comment by ferruzzi** [comment]
Sorry about that; you are right, I did forget to rebase.  Rebased, which required a force push, and added pr fixes commit.

FYI:  It looks like [this commit](https://github.com/spulec/moto/commit/76094f012bfbd90f583ef47e0ce313f631f00742) broke py2.7 compatibility with a syntax error at `moto/core/models.py::548`

File: tests/test_eks/test_eks.py (hunk 2 above)
Comment (ferruzzi) [reply tests/test_eks/test_eks.py]: Looks like the server mode tests don't like that:  https://github.com/spulec/moto/pull/4185/checks?check_run_id=3364472186#step:9:4609

Comment (bblommers) [reply tests/test_eks/test_eks.py:1042]: Mea culpa! I honestly thought that would work

**Review by bblommers — APPROVED** [review]
LGTM! Thanks again @ferruzzi

**Comment by ferruzzi** [comment]
I'm going to be taking some offline time, but afterward I'd like to fix up the EKS tests I contributed previously; I've learned better test-writing practices since that contrib so I'll clean those up and add some support for the EKS waiters I think.  We'll see.  Till then... take care 😁

**Comment by bblommers** [comment]
Awesome - tech debt PR's are always welcome 👍

**Pull request merged**